plugins/channel-safely/channel-safely-plugin.cpp

```diff
@@ -0,0 +1,269 @@
+            ChannelManager::Get().delete_groups();
+        }
+    }
+    // enable options
+    else if (parameters.size() == 1 && parameters[0] == "enable") {
+        return plugin_enable(out, true);
+    } else if (parameters.size() == 2 && parameters[0] == "enable" && parameters[1] == "cheats" && !cheat_mode) {
+        cheat_mode = true;
+        out.print("channel-safely: cheat mode enabled!\n");
+    } else if (parameters.size() == 2 && parameters[0] == "enable" && parameters[1] == "ontick" && !useTicks) {
+        useTicks = true;
+        out.print("channel-safely: ontick events enabled!\n");
+        if (enabled) {
+            EM::EventHandler tickHandler(onNewEvent, maxTickFreq);
+            if (!has_when<EM::EventHandler>::value) {
+                EM::registerTick(tickHandler, maxTickFreq, plugin_self);
+            } else {
+                EM::registerListener(EventType::TICK, tickHandler, plugin_self);
+            }
+        }
+    }
+    // disable options
+    else if (parameters.size() == 1 && parameters[0] == "disable") {
+        return plugin_enable(out, false);
+    } else if (parameters.size() == 2 && parameters[0] == "disable" && parameters[1] == "cheats" && cheat_mode) {
+        cheat_mode = false;
+        out.print("channel-safely: ontick events disabled!\n");
+    } else if (parameters.size() == 2 && parameters[0] == "disable" && parameters[1] == "ontick" && useTicks) {
+        useTicks = false;
+        out.print("channel-safely: ontick events disabled!\n");
+        if (enabled) {
+            EM::unregisterAll(plugin_self);
+            EM::EventHandler jobStartHandler(onStart, 0);
+            EM::EventHandler jobCompletionHandler(onComplete, 0);
+            EM::registerListener(EventType::JOB_INITIATED, jobStartHandler, plugin_self);
+            EM::registerListener(EventType::JOB_COMPLETED, jobCompletionHandler, plugin_self);
+        }
+    }
+    // developer debug
+    else if (parameters.size() == 1 && parameters[0] == "debug") {
+        debug_out = &out;
+        ChannelManager::Get().debug();
+        debug_out = nullptr;
+    } else if (parameters.size() == 1 && parameters[0] == "help") {
+        return CR_WRONG_USAGE;
+    } else {
+        debug_out = nullptr;
+        out.color(color_value::COLOR_RED);
+        out.print("Invalid argument.\n\n");
+        out.color(color_value::COLOR_RESET);
+        return CR_WRONG_USAGE;
+    }
+    debug_out = nullptr;
+    return CR_OK;
+}
+
+DFhackCExport command_result plugin_enable(color_ostream &out, bool enable) {
+    namespace EM = EventManager;
+#ifdef CS_DEBUG
+    debug_out = &out;
+    out.print("debugging enabled\n");
+#endif
+    if (enable && !enabled) {
+        using namespace EM::EventType;
+        EM::EventHandler tickHandler(onNewEvent, maxTickFreq);
+        EM::EventHandler jobStartHandler(onStart, 0);
+        EM::EventHandler jobCompletionHandler(onComplete, 0);
+        if (!has_when<EM::EventHandler>::value) {
```

Comment:
I suppose I should remove this now. Might save the template somewhere for my own future reference though.